packages/cli/src/util/deploy/create-git-meta.ts

```diff
@@ -0,0 +1,155 @@
+import fs from 'fs-extra';
+import { join } from 'path';
+import ini from 'ini';
+import git from 'git-last-commit';
+import {
+  BitbucketMeta,
+  GitHubMeta,
+  GitLabMeta,
+  GitMeta,
+  RepoData,
+} from '../../types';
+import { Output } from '../output';
+
+function getLastCommit(directory: string): Promise<git.Commit> {
+  return new Promise((resolve, reject) => {
+    git.getLastCommit(
+      (err, commit) => {
+        if (err) return reject(err);
+        resolve(commit);
+      },
+      { dst: directory }
+    );
+  });
+}
+
+export async function getRepoData(configPath: string, output: Output) {
+  let gitConfig;
+  try {
+    gitConfig = ini.parse(await fs.readFile(configPath, 'utf-8'));
+  } catch (error) {
+    output.debug(`Error while parsing repo data: ${error.message}`);
+  }
+  if (!gitConfig) {
+    return;
+  }
+
+  const originUrl = gitConfig['remote "origin"']?.url;
+  if (originUrl) {
+    return parseRepoUrl(originUrl);
+  }
+}
+
+export function parseRepoUrl(originUrl: string): RepoData | null {
+  const isSSH = originUrl.startsWith('git@');
+  // Matches all characters between (// or @) and (.com or .org)
+  const provider = originUrl.match(/(?<=(\/\/|@)).*(?=(.com|.org))/);
+  if (!provider) {
+    return null;
+  }
+
+  let org;
+  let repo;
+
+  if (isSSH) {
+    org = originUrl.split(':')[1].split('/')[0];
+    repo = originUrl.split('/')[1].replace('.git', '');
+  } else {
+    // Assume https:// or git://
+    org = originUrl.split('/')[3];
+    repo = originUrl.split('/')[4].replace('.git', '');
+  }
+
+  return {
+    provider: provider[0],
+    org,
+    repo,
+  };
+}
+
+export async function createGitMeta(
+  directory: string,
+  output: Output
+): Promise<GitMeta> {
+  const repoData = await getRepoData(join(directory, '.git/config'), output);
+  // If we can't get the repo URL, then don't return any metadata
+  if (!repoData) {
+    return {};
+  }
+  const commit = await getLastCommit(directory);
+
+  if (repoData.provider === 'github') {
```

Comment:
I'm not a huge fan of adding this logic to the cli because it means adding new providers requires cli updates. Although it happens rarely enough that it might be okay.

Reply:
My thinking is that the purpose of this feature is to populate data to be displayed on the deployments page of the dashboard, and that page specifically supports GitHub, Bitbucket, and GitLab with slightly different metadata for each one. If we wanted to support another provider, the dashboard would also have to be updated to support the new provider and the CLI would have to be updated anyway.

Reply:
The alternative is to let the deployment API parse the string so the CLI can just pass-through. This means that adding a new provider happens in 1 place: the API